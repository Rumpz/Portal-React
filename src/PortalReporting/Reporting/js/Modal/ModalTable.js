import React from 'react';
import './css/modalTable.css';

const ModalTable = (props) => {
  return (
    <div className='table table-hover'>
      <table>
        <Header arrValues={props.header} />
        <Body body={props.values} actions={props.actions} style={props.style} />
      </table>
    </div>
  );
};

const Header = ({arrValues}) => {
  let thVals = arrValues.map((value, idx) => {
    return <th key={idx}>{value}</th>;
  });
  return (
    <thead>
      <tr>{thVals}</tr>
    </thead>
  );
};

const Body = ({body, actions, style}) => {
  let row = body.values.map((value, idx) => {
    return <Row style={style} key={idx} value={value} actions={actions} />;
  });
  return (
    <tbody>{row}</tbody>
  );
};

const Row = ({value, actions, style}) => {
  let tRows = value.map((element, idx) => {
    return <Cell key={idx} value={element} />;
  });
  return (
    <tr onClick={actions ? actions.row.bind(null, value) : null}>{tRows}</tr>
  );
};

const Cell = ({value}) => {
  return <td>{value}</td>;
};

export default ModalTable;
