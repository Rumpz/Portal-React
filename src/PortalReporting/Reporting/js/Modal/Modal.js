import React from 'react';
import ModalTable from './ModalTable';
import ReactModal from 'react-modal';
import pasteAndGo from '../../../../assets/img/pasteAndGo.png';
import './css/modal.css';

const Modal = (props) => {
  const link = props.reportUrl.includes('http')
    ? <LinkedLink link={props.reportUrl} />
    : <PhisicalLink link={props.reportUrl} />;

  return (
    <div className='container'>
      <ReactModal
        appElement={props.body}
        isOpen={props.isOpen}
        className='Modal'
        overlayClassName='Overlay'>
        <div className='modal-body'>
          <h4>Relatório disponivel no link abaixo indicado</h4>
          <p>Os relatórios procurados podem ser extraidos através do link em baixo, excepto relatórios localizados na rede</p>
        </div>
        {link}
        <div className='modal-buttons'>
          <button
            className='btn btn-default'
            onClick={props.closeModal}>
            Close
          </button>
        </div>
      </ReactModal>
    </div>
  );
};
/* <ModalTable
header={props.modalHeader}
values={props.modalBody}
/> */
const LinkedLink = (props) => {
  return (
    <a href={props.link} target='_blank'>Download Link</a>
  );
};

const PhisicalLink = (props) => {
  return (
    <div>
      <strong>
        <p style={{fontSize: '12px'}}>
          Relatório encontra-se na share
          por motivos de segurança por favor
          copie o link e abra em outra janela
          <img alt='' style={{width: '50%', height: '50%'}}src={pasteAndGo} />
        </p>
      </strong>
      <strong>
        <p style={{fontSize: '12px'}}>Link:<br />
          <a>{props.link}</a>
        </p>
      </strong>
    </div>
  );
};

export default Modal;
