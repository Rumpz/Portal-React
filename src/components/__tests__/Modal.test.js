
import EnzymeAdapter from 'enzyme-adapter-react-16';
import ShallowRenderer from 'react-test-renderer/shallow';
import Enzyme from 'enzyme';
import React from 'react';
import Modal from '../Modal/Modal';

// This sets up the adapter to be used by Enzyme
Enzyme.configure({ adapter: new EnzymeAdapter() });

// Test Reports Component Render
// It should render an unordered list with the given data drray
it('renders reports list correctly', async () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Modal
    reportUrl={'Teste'}
    title={'Resumo do relatÃ³rio'}
    modalHeader={['testeHeader']}
    modalBody={['testBody']}
  />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
