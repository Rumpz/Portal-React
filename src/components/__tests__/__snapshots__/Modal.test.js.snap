// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders reports list correctly 1`] = `
<div
  className="container"
>
  <Modal
    appElement={undefined}
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="Modal"
    closeTimeoutMS={0}
    isOpen={false}
    overlayClassName="Overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="modal-body"
    >
      <h4>
        Resumo do relat√≥rio
      </h4>
      <h4>
        (possivelmente resumo do report em imagem ou dados em tabela)
      </h4>
      <ModalTable
        header={
          Array [
            "testeHeader",
          ]
        }
        values={
          Array [
            "testBody",
          ]
        }
      />
    </div>
    <PhisicalLink
      link="Teste"
    />
    <div
      className="modal-buttons"
    >
      <button
        className="btn btn-default"
        onClick={undefined}
      >
        Close
      </button>
    </div>
  </Modal>
</div>
`;
